rust/src/transformations/polars/make_select_trans/mod.rs

```diff
@@ -0,0 +1,273 @@
+use polars::prelude::*;
+
+use crate::core::{Function, StabilityMap, Transformation};
+use crate::domains::{Context, ExprDomain, LazyFrameDomain};
+use crate::error::*;
+use crate::metrics::{IntDistance, SymmetricDistance};
+
+/// Make a Transformation that applies list of transformations to a Lazy Frame.
+///
+pub fn make_select_trans(
+    input_domain: LazyFrameDomain,
+    input_metric: SymmetricDistance,
+    transformations: Vec<
+        Transformation<ExprDomain, ExprDomain, SymmetricDistance, SymmetricDistance>,
+    >,
+) -> Fallible<Transformation<LazyFrameDomain, LazyFrameDomain, SymmetricDistance, SymmetricDistance>> where
+{
+    let fist_transformation = transformations
+        .first()
+        .ok_or_else(|| err!(MakeTransformation, "transformation list cannot be empty"))?;
+
+    let output_domain = fist_transformation.output_domain.lazy_frame_domain.clone();
+
+    let output_metric = fist_transformation.output_metric.clone();
+
+    if transformations
+        .iter()
+        .any(|t| t.input_domain.lazy_frame_domain != input_domain)
+    {
+        return fallible!(MakeTransformation, "input domains do not match");
+    }
+
+    if transformations
+        .iter()
+        .any(|t| t.output_domain.context != Context::Select)
+    {
+        return fallible!(
+            MakeTransformation,
+            "transformation is not in select context"
+        );
+    }
+
+    if transformations
+        .iter()
+        .any(|t| t.output_domain.lazy_frame_domain != output_domain)
+    {
+        return fallible!(
+            MakeTransformation,
+            "transformations' output domains do not match"
+        );
+    }
+
+    let stability_maps: Vec<_> = transformations
+        .iter()
+        .map(|t| &t.stability_map)
+        .cloned()
+        .collect();
+
+    let stability_map = StabilityMap::new_fallible(move |d_in| {
+        let distances = stability_maps
+            .iter()
+            .map(|map| map.eval(d_in))
+            .collect::<Fallible<Vec<IntDistance>>>()?;
+
+        Ok(*distances.iter().max().expect("distances cannot be empty"))
+    });
+
+    Transformation::new(
+        input_domain.clone(),
+        output_domain,
+        Function::new_fallible(move |lazy_frame: &LazyFrame| -> Fallible<LazyFrame> {
+            let trans_outputs = transformations
+                .iter()
+                .map(|t| t.invoke(&(lazy_frame.clone(), all())))
```

Comment:
You could insert another map after this line to discard the frame:

```rust
    .map(|res| Ok(res?.1))
```

Then you don't need the second iter below.